.github/workflows/test.yml

```diff
@@ -95,6 +95,12 @@ jobs:
         name: errors
         path: Tests/errors
 
+    - name: Docs
+      if: startsWith(matrix.os, 'ubuntu') && matrix.python-version == 3.8
+      run: |
+        pip install sphinx-rtd-theme
+        make doccheck
+
     - name: After success
       if: success()
```

Comment:
Could you explain to me how `if: success()` is different to the default behaviour? I'm trying to figure out Docs could interact with the general flow.

Reply:
Yeah, actually, it looks like it doesn't do anything different in this context.

* Each yml file is a workflow.

  * Each workflow has a number of jobs. (We're only using a single job in our workflows.)

    * Each job has a number of steps.


The docs only mention steps:

> **success**
> Returns true when none of the previous steps have failed or been canceled.

https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#success

It's not entirely clear: perhaps it also checks previous steps in previous jobs?

Anyway, doesn't look like we actually need it.

Reply:
IIRC a previous version of the docs explicitly mentioned that `success()` is the default `if` condition. There is now only this: https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions

> You can use the following status check functions as expressions in if conditionals. If your if expression does not contain any of the status functions it will automatically result with success().

Reply:
Thanks for the explanations. I've created PR #4542 to remove them then, if they're not doing anything.